noop.go

```diff
@@ -68,7 +70,25 @@ func (n NoopTracer) StartSpanWithOptions(opts StartSpanOptions) Span {
 	return defaultNoopSpan
 }
 
+// Extractor belongs to the Tracer interface.
+func (n NoopTracer) Extractor(format interface{}) Extractor {
+	return defaultNoopInjectorExtractor
+}
+
+// Injector belongs to the Tracer interface.
+func (n NoopTracer) Injector(format interface{}) Injector {
+	return defaultNoopInjectorExtractor
+}
+
 // JoinTrace belongs to the Tracer interface.
 func (n NoopTracer) JoinTrace(operationName string, parent interface{}) Span {
 	return defaultNoopSpan
 }
+
+func (n noopInjectorExtractor) InjectSpan(span Span, carrier interface{}) error {
+	return nil
+}
+
+func (n noopInjectorExtractor) JoinTrace(operationName string, carrier interface{}) (Span, error) {
```

Comment:
Yeah, that's the way to go.

On Fri, Feb 12, 2016, 17:59 Yuri Shkuro <notifications@github.com> wrote:

> In noop.go
> <https://github.com/opentracing/opentracing-go/pull/48#discussion_r52810173>
> :
>
> >  // JoinTrace belongs to the Tracer interface.
> >  func (n NoopTracer) JoinTrace(operationName string, parent interface{}) Span {
> >  	return defaultNoopSpan
> >  }
> > +
> > +func (n noopInjectorExtractor) InjectSpan(span Span, carrier interface{}) error {
> > +	return nil
> > +}
> > +
> > +func (n noopInjectorExtractor) JoinTrace(operationName string, carrier interface{}) (Span, error) {
>
> +1 to std error
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opentracing/opentracing-go/pull/48/files#r52810173>.
>